src/app/routines/publicRoutines/routineListItem/RoutineListItem.tsx

```diff
@@ -3,34 +3,98 @@ import { ListItem, ListItemText, Typography } from '@mui/material';
 import './RoutineListItem.scss';
 import { useNavigate } from 'react-router-dom';
 import { AppRoutes } from '@/types/routes';
+import ThumbUpOffAltIcon from '@mui/icons-material/ThumbUpOffAlt';
+import { queryClient } from '@/config/tanstack_query/config';
+import { useMutation } from '@tanstack/react-query';
+import { like } from '@/api/routineLikeApi';
+import { useState } from 'react';
+import AppAlert from '@/app/components/alerts/AppAlert';
 
 const RoutineListItem = ({ routine }: { routine: Routine }) => {
-  const { id = '', name = 'No name', description = 'No description' } = routine;
+  const {
```

Comment:
If routine is completely missing (e.g., undefined), this will throw an error because destructuring cannot be performed on undefined. To handle that, you could provide a fallback object, like this:

const {
  id = '',
  name = 'No name',
  description = 'No description',
  likesCount = 0,
  isLikedByCurrentUser = false,
} = routine || {};